tests/test_other.py

```diff
@@ -8526,11 +8521,19 @@ def test_check_sourcemapurl(self):
   def test_check_sourcemapurl_default(self):
     if not self.is_wasm():
       self.skipTest('only supported with wasm')
-    run_process([PYTHON, EMCC, path_from_root('tests', 'hello_123.c'), '-g4', '-o', 'a.js'])
-    output = open('a.wasm', 'rb').read()
-    # has sourceMappingURL section content and points to 'a.wasm.map' file
-    source_mapping_url_content = encode_leb(len('sourceMappingURL')) + b'sourceMappingURL' + encode_leb(len('a.wasm.map')) + b'a.wasm.map'
-    self.assertIn(source_mapping_url_content, output)
+
+    def test(args):
+      try_delete('a.wasm.map')
+      run_process([PYTHON, EMCC, path_from_root('tests', 'hello_123.c'), '-g4', '-o', 'a.js'] + args)
+      output = open('a.wasm', 'rb').read()
+      # has sourceMappingURL section content and points to 'a.wasm.map' file
+      source_mapping_url_content = encode_leb(len('sourceMappingURL')) + b'sourceMappingURL' + encode_leb(len('a.wasm.map')) + b'a.wasm.map'
+      self.assertIn(source_mapping_url_content, output)
+
+    test([])
+    # we should still emit a source map if other options alter the debug level
+    # (see #8584)
+    test(['--profiling'])
```

Comment:
This sounds like a good opportunity to use `@parameterized`.

Reply:
Thanks, done with your help.